src/kixi/datastore/transport_specs.clj

```diff
@@ -16,14 +16,33 @@
   (s/keys :req [::ms/id ::ms/size-bytes ::ms/provenance]))
 
 (def file-type->default-metadata
-  {"csv" {::ms/headers true}})
+  {"csv" {::ms/header true}})
 
 (def default-primary-metadata
   {::ms/type "csv"})
 
 (s/fdef filemetadata-transport->internal
         :args (s/cat :meta ::filemetadata-transport
                      :details ::file-details)
+        :fn #(let [{:keys [meta details]} (get % :args)
+                   file-metadata (get % :ret)]
+               (and (= (::ms/id details)
+                       (::ms/id file-metadata))
+                    (= (::ms/size-bytes details)
+                       (::ms/size-bytes file-metadata))
+                    (= (::ms/provenance details)
+                       (::ms/provenance file-metadata))
+                    (= (:schema-id meta)
+                       (::ss/id file-metadata))
+                    (= (:name meta)
+                       (::ms/name file-metadata))
+                    (case (or (:type meta) 
+                              (::ms/type file-metadata)) 
+                      "csv" (if-not (nil? (:header meta))
```

Comment:
It should be being explored by the generator for the ::ms/type in the transport spec and it's optional. So, I think it should be exploring both the 'or' possibilities.